src/app/public/modules/lookup/lookup.component.ts

```diff
@@ -113,6 +121,33 @@ export class SkyLookupComponent
   @Input()
   public idProperty: string;
 
+  /**
+   * Specifies the selection mode that determines whether users can select one item or
+   * multiple items. The valid options are `single` and `multiple`.
+   * @default "mulitple"
+   */
+  @Input()
+  public set selectMode(value: SkyLookupSelectMode) {
+    let multipleToSingle: boolean = value === SkyLookupSelectMode.single && this.selectMode === SkyLookupSelectMode.multiple;
+
+    this._selectMode = value;
+    this.updateForSelectMode();
+
+    if (multipleToSingle) {
+      if (this.tokens && this.tokens.length > 1) {
+        // The `setTimeout` is needed to avoid a `ExpressionChangedAfterItHasBeenCheckedError` error in template forms.
+        setTimeout(() => {
+          this.writeValue([this.tokens[0].value]);
```

Comment:
Is selecting the first value a requirement? Seems like we're assuming the consumer will always want to select the first one... Changing the select mode is a pretty fundamental change to how the component works - I think one could argue that we should just clear out the value completely. It would also make your code a lot simpler. What do you think?

Reply:
I have no strong preference on that. I did that because we were just leaving the value as is and it was putting the component into a state where it was in single select mode but multiple values were selected. We don't clear when going from single to multiple because the single value is valid in that mode. Maybe something we should chat about in flex?